plugin/trino-accumulo/src/main/java/io/trino/plugin/accumulo/AccumuloConnectorFactory.java

```diff
@@ -42,6 +43,7 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
         requireNonNull(catalogName, "catalogName is null");
         requireNonNull(config, "config is null");
         requireNonNull(context, "context is null");
+        checkSpiVersion(context, this);
```

Comment:
Shouldn't this be called in the `PluginManager`? Third-party plugins won't be checked by default unless they extend Trino plugins, right? In what situations Trino plugins could be mixed with different versions of the Trino server?

Reply:
> Shouldn't this be called in the PluginManager? 

Doing the check the other way around would prevent building a plugin that's supposed to work with multiple Trino versions. Or, we would need the mechanism to be more complicated (declare range of supported versions, or someting).

> Third-party plugins won't be checked by default 

Correct.

> unless they extend Trino plugins, right?

Correct. Note however, that if you extend from base-jdbc, the SPI and base-jdbc versions will be used for the check, not your plugin's version.

> In what situations Trino plugins could be mixed with different versions of the Trino server?

If the author of the plugin chooses to support multiple Trino versions.
Trino project doesn't choose so for the plugins/connectors maintained by the Trino project.

The change only codifies what we always knew and were telling the users: one must not use different server and plugin versions (for plugins supported by us).

Reply:
Thanks for the explanation. Looks like I've made a wrong assumption about the purpose of this. Maybe this warrants adding a comment to `checkSpiVerrsion` explaining when it should be used and the side effects of that?

Reply:
Sure, can you please suggest a wording?

Reply:
I added this as a separate comment for the `checkSpiVersion` method definition.